src/main/java/bio/terra/service/resourcemanagement/azure/AzureContainerPdao.java

```diff
@@ -20,34 +21,62 @@ public AzureContainerPdao(AzureAuthService authService) {
    * @param storageAccountResource Metadata describing the storage account that contains the
    *     container
    * @param containerType The nature of the container
    * @return A client connection object that can be used to create folders and files
    */
-  public DataLakeFileSystemClient getOrCreateContainer(
+  public BlobContainerClient getOrCreateContainer(
       BillingProfileModel profileModel,
       AzureStorageAccountResource storageAccountResource,
       AzureStorageAccountResource.ContainerType containerType) {
-    DataLakeServiceClient dataLakeClient =
-        authService.getDataLakeClient(profileModel, storageAccountResource);
-
-    try {
-      DataLakeFileSystemClient fileSystemClient =
-          dataLakeClient.getFileSystemClient(
-              storageAccountResource.determineContainer(containerType));
-      // This will 404 if the container doesn't exist
-      fileSystemClient.getProperties();
-      return fileSystemClient;
-    } catch (DataLakeStorageException e) {
-      if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
-        return dataLakeClient.createFileSystem(
+    BlobContainerClient blobContainerClient =
+        authService.getBlobContainerClient(
+            profileModel,
+            storageAccountResource,
             storageAccountResource.determineContainer(containerType));
-      }
-      throw new PdaoException("Could not create a Storage account container");
+
+    if (!blobContainerClient.exists()) {
+      blobContainerClient.create();
     }
+    return blobContainerClient;
+  }
+
+  public String getDestinationContainerSignedUrl(
+      BillingProfileModel profileModel,
+      AzureStorageAccountResource storageAccountResource,
+      AzureStorageAccountResource.ContainerType containerType,
+      boolean enableRead,
+      boolean enableList,
+      boolean enableWrite,
+      boolean enableDelete) {
+
+    BlobContainerSasPermission permissions =
+        new BlobContainerSasPermission()
+            .setReadPermission(enableRead)
+            .setWritePermission(enableWrite)
+            .setListPermission(enableList)
+            .setDeletePermission(enableDelete);
+
+    OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+    SasProtocol sasProtocol = SasProtocol.HTTPS_ONLY;
+
+    // build the token
+    BlobServiceSasSignatureValues sasSignatureValues =
+        new BlobServiceSasSignatureValues(expiryTime, permissions)
+            .setProtocol(sasProtocol)
+            // Version is set to a version of the token signing API the supports keys that permit
+            // listing files
+            .setVersion("2020-04-08");
+
+    BlobContainerClient containerClient =
+        getOrCreateContainer(profileModel, storageAccountResource, containerType);
+    return String.format(
+        "%s?%s",
+        containerClient.getBlobContainerUrl(),
+        containerClient.generateSas(sasSignatureValues, new Context("FOO", "BAR")));
```

Comment:
Placeholder code?

Also, we should be using URL APIs to add query parameters to a URL, so proper quoting is taken care of.

Reply:
Ah shoot, you're totally right.  Sorry, this was when I was trying to figure out some stuff with encoding information in Sas tokens

Reply:
Normally I'd agree but the way that the Azure API work it returns a Sas token which is a URL-ready set of URL query parameters so adding to a URL will double encode (or we need to do a bunch of extra processing, which doesn't really buy us much)

Reply:
> Normally I'd agree but the way that the Azure API work it returns a Sas token which is a URL-ready set of URL query parameters so adding to a URL will double encode (or we need to do a bunch of extra processing, which doesn't really buy us much)

OK, it wasn't clear to me from looking at the code that the query argument was already encoded. It's a shame the container client itself doesn't have a way to make URLs. Maybe this is the wrong place to construct a URL? It looks like other uses of this API are in `BlobContainerClientFactory`, e.g. `createReadOnlySasUrlForBlob()`.